src/parser/core/modules/PrecastStatus.ts

```diff
@@ -0,0 +1,153 @@
+import Module, {dependency} from 'parser/core/Module'
+import {Event} from 'events'
+import {Data} from 'parser/core/modules/Data'
+import {Status} from 'data/STATUSES'
+import {AbilityEvent, CastEvent} from 'fflogs'
+
+const trackedBuffEventTypes = ['applybuff', 'applybuffstack', 'removebuff', 'removebuffstack', 'refreshbuff']
+const isTrackedBuffEvent = (event: Event): event is AbilityEvent => (trackedBuffEventTypes.includes(event.type as string))
+// Statuses applied before the pull won't have an apply(de)?buff event
+// Fake buff applications so modules don't need to take it into account
+export class PrecastStatus extends Module {
+	static handle = 'precastStatus'
+	static debug = true
+
+	@dependency private data!: Data
+
+	trackedStatuses = new Map<number, number[]>()
+	trackedActions: number[] = []
+	buffEventsToSynth: AbilityEvent[] = []
+	castEventsToSynth: CastEvent[] = []
```

Comment:
these seem like internal details of how the module works - should they be private?